iterators.tex

```diff
@@ -276,7 +276,7 @@
   template <class I>
   concept bool Readable@\newtxt{() \{}\oldtxt{ =}@
     @\newtxt{return}@ Semiregular<I>@\newtxt{()}@ &&
-      requires (I i) {
+      requires (@\newtxt{const }@I i) {
```

Comment:
> I also wonder if the committee will balk at our defining "semantically equivalent" to allow for observably different side-effects.

Almost certainly yes. The "as-if" rule is the closest thing to semantic equivalence in the standard, and it covers all observable effects. This weaker definition probably can't be applied usefully to the rest of the standard. I think instead of trying to define "semantically equivalent" to be what I need for "implicit expression variants", I should explicitly characterize the requirements for expression variants. What I really need is substitutability: the expression variants must satisfy the requirements of the primary expression. A couple of sentences to describe that is probably a more feasible goal than trying to generally define "semantically equivalent."